src/renderer/psp/js/index.js

```diff
@@ -0,0 +1,58 @@
+/*!
+Copyright 2017 Raising the Floor - International
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+The research leading to these results has received funding from the European Union's
+Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+/* global fluid */
+
+"use strict";
+(function (fluid) {
+    var gpii = fluid.registerNamespace("gpii");
+    fluid.registerNamespace("gpii.psp");
+
+    fluid.defaults("gpii.psp", {
+        gradeNames: ["fluid.component"],
+        components: {
+            clientChannel: {
+                type: "gpii.psp.clientChannel",
+                options: {
+                    listeners: {
+                        onPreferencesUpdated: {
+                            funcName: "{mainWindow}.updatePreferences"
+                        },
+                        onAccentColorChanged: {
+                            funcName: "{mainWindow}.updateTheme"
+                        },
+                        onSettingUpdated: {
+                            funcName: "{mainWindow}.updateSetting"
+                        }
+                    }
+                }
+            },
+
+            mainWindow: {
+                type: "gpii.psp.mainWindow",
+                container: "#flc-body",
+                options: {
+                    listeners: {
+                        onPSPClose: "{clientChannel}.close",
+                        onKeyOut: "{clientChannel}.keyOut",
+                        onSettingAltered: "{clientChannel}.alterSetting",
+                        onActivePreferenceSetAltered: "{clientChannel}.alterActivePreferenceSet",
+                        onContentHeightChanged: "{clientChannel}.changeContentHeight"
+                    }
+                }
+            }
+        }
+    });
+
+    $(function () {
```

Comment:
It may seem perverse, but I'd like to see this function invocation moved into a 1-line <script tag at the base of the markup instead. It helps reuse a lot if one can separate the process of grade definition and instantiation into different files.

Reply:
I moved the definition of the `gpii.psp` component to a separate file and left its instantiation here. This is equivalent to putting a script tag at the end of the body tag in the markup. I always refrain from placing JavaScript code directly in a script tag within an HTML file unless it is absolutely necessary. In this particular case, I do not think that such a necessity is present.